affinityworks/main
# resolve "Per group feature toggle" (#702)

Closes #702 

# Functionality

Add group-by-group feature toggles with following API:

## Discovering Toggles

```ruby
FeatureToggle::FEATURES
```

## Inspecting All Toggles for a Group

```ruby
Group.find(3).feature_toggles
```

## Inspecting One Toggle for a Group

```ruby
Group.find(3).feature_on?(:email_google_group)
```

## Changing One Toggle for a Group

```ruby
Group.find(3).toggle_feature_on(:email_google_group)
Group.find(3).toggle_feature_off(:email_google_group)
```

## Generate feature toggle with Rake task

```shell
$ rake feature_toggles:create[feature_name]
```

# QA Notes
1. Find a group that has a google group by doing `GoogleGroup.last.group.id` in rails console
2. Go to /groups/id/dashboard in browser where id is result from 1
3. VERIFY that you DO NOT see the email google group link
4. Go into rails console and run `Group.find(id).toggle_feature_on(:email_google_group)`
5. Refresh the page in browser
6. VERIFY that you DO see the email google group link
7. Drink beer 🍻
